### PR TITLE
feat: implement "local" hooks

### PR DESCRIPTION
**Pull request by isidentical**
Implements the ability to specify the local (user's own environment) for hooks.

### CODE REVIEW

File: src/faldbt/parse.py [hunk 1]
```diff
@@ -174,22 +174,27 @@ def get_global_script_configs(source_dirs: List[Path]) -> Dict[str, List[str]]:
...
-    else:
-        raise FalParseError(
-            f"Could not find the environment definition for {name} in {fal_project_path}"
-        )
+        if "name" not in environment:
+            raise FalParseError("Environment must have a name")
+
+        name = environment.get("name")
+        if _is_local_environment(name):
+            raise FalParseError(f"Environment name conflicts with a reserved name: {name}.")
```
Comment (chamini2) [inline src/faldbt/parse.py:194]: How do we feel of making the `local` environment actually a `null`?

https://www.educative.io/answers/how-to-represent-null-values-in-yaml

```yaml
        post-hook:
          - path: environment_type_2.py
            environment: null
```


```yaml
        post-hook:
          - path: environment_type_2.py
            environment: ~
```


```yaml
        post-hook:
          - path: environment_type_2.py
            environment:
```

Comment (chamini2) [reply src/faldbt/parse.py:194]: I do like how `local` is explicit. But I don't know, just a thought.

**Review by chamini2 — APPROVED** [review]
preeetty cool

File: src/faldbt/parse.py (hunk 1 above)
Comment (isidentical) [reply src/faldbt/parse.py:194]: @mederka @turbo1912 @burkaygur any preferences? It is kind of annoying that this prevents people from naming their own environments `local` but I feel just like @chamini2 in the sense that it is more explicit than `~`.

Comment (isidentical) [reply src/faldbt/parse.py:194]: Let's merge this as is, and if anyone has any extra thoughts on this we can address them separately on https://linear.app/features-and-labels/issue/FEA-372/decide-on-local-vs-for-local-environments.

**Pull request merged**